sources/api/apiserver/src/server/controller.rs

```diff
@@ -443,19 +592,164 @@ pub(crate) fn get_metadata_for_all_data_keys<D: DataStore, S: AsRef<str>>(
+                .context(error::DeserializeSettingsGeneratorSnafu)?;
+
+            trace!(
+                "Getting the metadata for key {:?} with setting generator {:?}",
+                data_key,
+                setting_generator
+            );
+            expand_setting_generator(datastore, &data_key, setting_generator, &mut result)?;
         }
     }
+
+    Ok(result)
+}
+
+/// Gets the value of a metadata key everywhere it's found in the data store.  Returns a mapping
+/// of data key to the metadata Key and value associated with the requested key.
+pub(crate) fn get_metadata_for_all_data_keys<D: DataStore, S: AsRef<str>>(
+    datastore: &D,
+    md_key_str: S,
+) -> Result<HashMap<Key, HashMap<Key, String>>> {
+    trace!("Getting metadata '{}'", md_key_str.as_ref());
+    let result = datastore
+        .get_metadata_prefix("", &Committed::Live, &Some(md_key_str))
+        .context(error::DataStoreSnafu {
+            op: "get_metadata_prefix",
+        })?;
+
     Ok(result)
 }
 
+// Parses and validates the settings and metadata in pending transaction and
+// returns the constraint check result containing approved settings and metadata to
+// commit to live transaction.
+// We will pass this function as argument to commit transaction function.
+fn datastore_transaction_check<D>(
+    datastore: &mut D,
+    committed: &Committed,
+) -> Result<ConstraintCheckResult>
+where
+    D: DataStore,
+{
+    // Get settings to commit from pending transaction
+    let settings_to_commit = datastore
+        .get_prefix("settings.", committed)
+        .context(error::DataStoreSnafu { op: "get_prefix" })?;
+
+    // Get metadata from pending transaction
+    let mut transaction_metadata = datastore
+        .get_metadata_prefix("settings.", committed, &None as &Option<&str>)
+        .context(error::DataStoreSnafu {
+            op: "get_metadata_prefix",
+        })?;
+
+    let mut metadata_to_commit: Vec<(Key, Key, String)> = Vec::new();
+
+    // Parse and validate all the metadata enteries from pending transaction
+    for (key, value) in transaction_metadata.iter_mut() {
+        for (metadata_key, metadata_value) in value {
+            // For now we are only processing the strength metadata from pending
+            // transaction to live
+            if metadata_key.name() != "strength" {
+                warn!(
+                    "Metadata key is {}, Skipping the commit as we only allow Strength metadata.",
+                    metadata_key.name()
+                );
+                continue;
+            }
+
+            // strength in pending transaction
+            let pending_strength: Strength =
+                deserialize_scalar::<String, ScalarError>(metadata_value.as_ref())
+                    .with_context(|_| error::DeserializeStrengthSnafu {})?
+                    .parse::<Strength>()
+                    .context(error::ParseStrengthSnafu)?;
```

Comment:
Calling `.parse()` uses `FromStr` but `deserialize_scalar` uses `serde` directly, so we probably don't need to parse to a String first.

```suggestion
            let pending_strength =
                deserialize_scalar::<Strength, ScalarError>(metadata_value.as_ref())
                    .with_context(|_| error::DeserializeStrengthSnafu {})?;
```